### ISSUE TITLE
Cython API for special.spherical_jn

### ISSUE THREAD

**Issue by pisarik** [scipy/scipy#11029]
Hi,

I want to call `scipy.special.spherical_jn` function directly from Cython (i.e. without gil).
But unfortunately I can't find it in the list of Cython API for the special functions.

Inspecting source code of scipy I have found a `_spherical_bessel.pxd` with nogil functions like [`spherical_jn_real`](https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/scipy/special/_spherical_bessel.pxd#L74) and others, so it looks like there should be way to call them without gil, but I can't find anything about it in my pip build of scipy v1.3.1.

Is it possible to call `scipy.special.spherical_jn` from Cython without gil?

**Comment by person142** [comment]
Hey @pisarik, the reason they aren’t there is because the spherical Bessel function signatures have a Python wrapper to add a keyword argument (`derivative`) and the current Cython special codegen can’t handle that automatically.

I’ve been meaning to hand-write some Cython wrappers for situations like this (`zeta` is another one).

As a stopgap, you could maybe use the relation to the Bessel functions?

**Comment by pisarik** [comment]
Hi @person142, thanks for super fast answer!:)

I think I am using right now stopgag you mentioned: `jv(i + 0.5, x) * sqrt(M_PI_2 / x)`.
But it turns out it works very slowly comparing to `spherical_jn`.

I have highly numpy vectorized function code just in Python, which gives me next timings:
1) Python code with `jv`: 700 ms
2) Cython code with `jv`: 680 ms
3) Cython code with `jv` + openmp: 145 ms
4) Python code with `spherical_jn`: 393 ms
5) Cython code with `spherical_jn`: 1.66 s

And this part of code is really a bottleneck, so improvement almost by factor 2 is very cool.

But what if I don't need this `derivative` addition? Can I call somehow `spherical_jn_real` that called from `_spherical_bessel.pxd`?

**Comment by pisarik** [comment]
Oh, I see. If the argument `x < n` then it uses just `cbesj` routine. But for big enough `x` it uses recurrent approach. I think I can just derive this part of `_spherical_bessel.pxd` as a stopgag. 

But if there is a way to call it I would use it :)

**Comment by pisarik** [comment]
Dear @person142,

I have found the article [Recurrence Techniques for the Calculation of Bessel Functions](https://www.ams.org/journals/mcom/1959-13-066/S0025-5718-1959-0105794-5/S0025-5718-1959-0105794-5.pdf) By M. Goldstein and R. M. Thaler very helpful for fast calculation of jv through recurrent formula with saving accuracy.

Also during working through the code in `_special_bessel.pxd` I have found that function `spherical_jn_complex` doesn't use recurrent formulas at all, but it can use them:)
